rules/builds.smk

```diff
@@ -10,6 +10,15 @@ rule download:
         aws s3 cp s3://nextstrain-ncov-private/sequences.fasta.gz - | gunzip -cq > {output.sequences:q}
         """
 
+rule combine_exclude_files:
+    message: "Combining the local `exclude.txt` with the generated one on S3"
+    output:
+        exclude = "config/exclude.txt"
+    shell:
+        """
+        aws s3 cp s3://nextstrain-ncov-private/exclude.txt - >> {output.exclude:q}
+        """
```

Comment:
Instead of modifying the hardcoded _config/exclude.txt_ in place, this rule would be better if it took `config["files"]["exclude"]` as its input and produced a separate file as its output, something like `results/exclude.txt` maybe. Then the `filter` rule immediately following would be adjusted to take `rules.combine_exclude_files.output.exclude` as its input.

This avoids modifying a git-tracked file, which will affect the git repo's dirty/clean status, and avoids hardcoding the input base excludes file.

The command should probably also make sure when it appends the remote excludes file that there's a trailing newline in the base file. I've noticed that not everyone's editor ensures this and some files are missing the trailing newline (which breaks a lot of assumptions that Unix tools make).

Reply:
Updated in the latest commit